### ISSUE TITLE
scroll functionality for IconButton only works within an application

### ISSUE THREAD

**Issue by stevenmos** [solus-project/budgie-desktop#1711]
**Operating System**: Solus 3.9999
**Budgie version** :budgie-desktop 10.5 (git-aa0d4797ed3d630710afa0c4d7e3d672f317222a)

**The issue encountered**

Since the latest update and, presumably, 2ff39f660f96641e0a66685f5d1db5447b3ce75b, the scroll functionality for IconButton only works for an application if there is more than one instance of it running.

Even if I only had one instance of several different programs running, I liked to bring one of them to the fore with a scroll rather than a click (I also use focus follows mouse).

**Steps to reproduce the issue**

- Maximise a Firefox window and two terminal windows and `hide' Firefox behind a terminal
- Firefox cannot be brought forward via scroll
- but the other terminal window can be

Moreover, there seems to be a bit of an inconsistency as to when one can use the scroll, even for programs with multiple instances e.g. (using the same example) if all three windows are minimised the terminals can't be brought up with a scroll but if Firefox is clicked (and maximised) scroll works for the terminals.
 ...

**Comment by JoshStrobl** [comment]
@stevenmos So just so I am understanding correctly, are you saying that you expect the behavior to be that if you only have a single window open of a single type (so for instance, only one Firefox window), that it would activate / be brought into focus on scroll up?

If that is the case, I have just implemented functionality to address exactly that. As of 5b6a5c534230b4b940364d8d19d8838766d5431c, if you scroll up on an IconButton which only has one window, it will now activate and be brought into focus. So if you had a window already showing, say Terminal, on another screen, it would be in focus. But if you had a window in the background and scroll up, it'd be put into the foreground.

Additionally I implemented it so if you scroll down, it will minimize that single window.

If this is the functionality you expect, please let me know so I can mark this as resolved :slightly_smiling_face:

**Comment by stevenmos** [comment]
Yes, thank you @JoshStrobl.  Hopefully this will also resolve the inconsistency I mention ("Moreover...") but if not I'll submit another issue.

**Comment by JoshStrobl** [comment]
> Moreover, there seems to be a bit of an inconsistency as to when one can use the scroll, even for programs with multiple instances e.g. (using the same example) if all three windows are minimised the terminals can't be brought up with a scroll but if Firefox is clicked (and maximised) scroll works for the terminals.

During my testing I have been unable to reproduce this issue. Steps I performed in attempt to reproduce:

1. Opened up 3 separate Terminal windows
2. Minimized each Terminal window
3. Scrolled up to go to the next window

The behavior of it then activating and showing worked in both cases where there was a window in front as well as no windows maximized on such screen.

**Comment by JoshStrobl** [comment]
Thanks for walking me through it. I'm able to reproduce the issue and I'll look into fixing it :+1:
